### PR TITLE
WIP sync from checkpoint

### PR DESCRIPTION
**Pull request by michaelneuder**
https://github.com/ledgerwatch/erigon/issues/5965. This PR starts the implementation of syncing from a checkpoint to head.

### CODE REVIEW

**Comment by michaelneuder** [comment]
OK got this working! Now `cmd/erigon-cl/main.go` does a few things. 

1. fetches the checkpoint state. (this is what Guilio has already written)
2. starts a sentinel service
3. uses the sentinel to fetch the checkpoint block by root

The next steps will be to  (1) ask for the status object from a peer, (2) use the status object to determine how many blocks to fetch by range (3) save those blocks to the DB.

**Comment by michaelneuder** [comment]
sample successful output

```
go run main.go
# command-line-arguments
ld: warning: -no_pie is deprecated when targeting new OS versions
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame
WARN[11-17|10:52:09.914] no log dir set, console logging only
INFO[11-17|10:52:09.928] [Checkpoint Sync] Requesting beacon state uri=https://mainnet-checkpoint-sync.stakely.io/eth/v2/debug/beacon/states/finalized
INFO[11-17|10:52:16.711] Checkpoint sync successful: hurray!
INFO[11-17|10:52:16.722] Starting sync from checkpoint.
INFO[11-17|10:52:16.722] Fork digest                              data="[74 38 197 139]"
INFO[11-17|10:52:16.722] [Sentinel] running sentinel with configuration cfg="&{GenesisCfg:0xc002938510 BeaconCfg:0xc0029da000 NetworkCfg:0xc00094b340 Port:8080 Addr:127.0.0.1 ServerAddr:localhost:7777 ServerProtocol:tcp ServerTcpPort:0 LogLvl:3 NoDiscovery:false CheckpointUri:https://mainnet-checkpoint-sync.stakely.io/eth/v2/debug/beacon/states/finalized}"
INFO[11-17|10:52:16.737] Sentinel started                         enr=enr:-Lq4QIZEAmsVnrObMBKgZOMVZCXNlX1zagZYsoIsXM-roRXYbm01h19R64IO8U6wdOgLqyHyXmfgLPBKb2kNuzE53PMBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpBKJsWLAgAAAP__________gmlkgnY0gmlwhH8AAAGJc2VjcDI1NmsxoQKS3ABm_7FMTZ3yGGinTYUSBpfebGMz3mgYHJYIlZXoxIhzeW5jbmV0cwCDdGNwgIN1ZHCCH5A
INFO[11-17|10:52:16.738] Sentinel started                         addr=localhost:7777
INFO[11-17|10:52:17.738] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
EROR[11-17|10:52:17.740] Error received from peer                 error="rpc error: code = Unknown desc = failed to connect to a random peer err=no peers"
INFO[11-17|10:52:18.738] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:19.738] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:20.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:21.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:22.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:23.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:24.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:25.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:26.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:27.737] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:28.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:29.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:30.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:31.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:32.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:33.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:34.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:35.736] Getting checkpoint block by root         root=ad05715896c7361fcb9f2dd175753d3e6d5d2889de9c95edc775f5e7f10e3b03
INFO[11-17|10:52:35.766] Block received by root
INFO[11-17|10:52:35.766] Retrieved root block.                    block="&{Block:0x4c00e8c06c0 Signature:[149 159 63 84 153 64 67 171 77 137 250 112 205 180 24 230 88 41 43 63 22 155 254 57 69 27 75 141 91 249 232 0 107 137 121 121 79 87 70 254 97 162 84 1 39 141 113 74 19 201 142 89 137 61 115 235 145 6 217 1 142 147 159 173 172 239 35 142 107 20 69 160 159 229 204 133 189 137 91 47 84 19 242 96 70 4 85 91 109 131 130 234 244 72 197 222]}"
```

**Comment by michaelneuder** [comment]
Screen Shot 2022-11-17 at 11 14 06 AM

File: cmd/erigon-cl/main.go [hunk 1]
```diff
@@ -2,19 +2,96 @@ package main
...
+	"github.com/ledgerwatch/erigon/cl/rpc/consensusrpc"
 	clcore "github.com/ledgerwatch/erigon/cmd/erigon-cl/cl-core"
 	cldb "github.com/ledgerwatch/erigon/cmd/erigon-cl/cl-core/cl-db"
+	lcCli "github.com/ledgerwatch/erigon/cmd/sentinel/cli"
+	"github.com/ledgerwatch/erigon/cmd/sentinel/cli/flags"
+	"github.com/ledgerwatch/erigon/cmd/sentinel/sentinel"
+	"github.com/ledgerwatch/erigon/cmd/sentinel/sentinel/service"
+	sentinelapp "github.com/ledgerwatch/erigon/turbo/app"
 	"github.com/ledgerwatch/log/v3"
+	"github.com/urfave/cli"
```
Comment (Giulio2002) [inline cmd/erigon-cl/main.go:21]: should be `	"github.com/urfave/cli/v2"`

**Review by Giulio2002 — APPROVED** [review]

**Comment by Giulio2002** [comment]
I did not mean to approve it lol, but I think we can merge it as a prototype, we ideally want to make asynchrounous spamming of requests to our peers work under a template but that can be kept for later :)

File: cmd/erigon-cl/main.go (hunk 1 above)
Comment (michaelneuder) [reply cmd/erigon-cl/main.go]: done

**Pull request merged**